### DISCUSSION TITLE
Is there a workaround for instantiating fixtures when developing within unit tests?

### DISCUSSION THREAD

**Discussion by mikewojnowicz** [pytest-dev/pytest#11484] [Q&A]
When I'm developing, I would like to write test code that looks like this:

```
def test_my_function(argument1, argument2):

    result = my_function(argument1, argument2) 

    breakpoint()
```
and then interact with the arguments of `test_my_function`.

In particular, after the `breakpoint()` launches the python debugger, I enter ipython, preserving contents of memory, via a command `ipy`, which is aliased in my `.pdbc` to

```
alias ipy from IPython import embed; user_ns=locals(); user_ns.update(globals()); embed(user_ns=user_ns)
```

This workflow lets me interact with the arguments of `test_my_function` -- e.g. to check their type and dimensionality, experiment with various transformations, etc.  

The problem is that when `argument1` and `argument2` are pytest fixtures,  my workflow is broken (I believe as of pytest v6.0.0).  Previously, I could write `argument1=argument1()` and interact with `argument1`.  Now I cannot.

Some questions:

- Is there some other way to accomplish what I'm trying to do?   
- How is it that other people have not run into a similar problem?  To me it seems "natural" to develop by writing unit tests and interacting with the arguments of the test in order to develop further.  Pytest's blocking of this strikes me as a fundamental problem.

**Comment by The-Compiler** [comment]
I'd suggest just running the test via pytest instead. If you're using IPython, you might want to use [ipytest](https://github.com/chmp/ipytest) to do so.
